File: 2643-row-with-maximum-ones/2643-row-with-maximum-ones.js
/**
 * @param {number[][]} mat
 * @return {number[]}
 */
var rowAndMaximumOnes = function(mat) {
    let idx = null, cnt = 0
    const map = {}
    for (const r in mat) {
        for (const c in mat[r]) {
            if (mat[r][c] === 1) {
                map[r] = map[r] + 1 || 1
                if (idx === null || map[r] > cnt) {
                    idx = r
                    cnt = map[r]
                }
            }
        }
    }
    return [idx, cnt]
};

rowAndMaximumOnes = function(mat) {
    return mat.reduce((acc, row, i) => {
        const [idx, cnt] = acc
        const sum = row.reduce((acc, val) => acc + val, 0)
        if (idx === null || sum > cnt) {
            return [i, sum]
        } else {
            return acc
        }
    }, [null, 0])
}

File: 2643-row-with-maximum-ones/2643-row-with-maximum-ones.py
class Solution:
    def rowAndMaximumOnes(self, mat: List[List[int]]) -> List[int]:
        res = [None, 0]
        for r, row in enumerate(mat):
            sm = sum(row)
            if res[0] == None or sm > res[1]:
                res[0] = r
                res[1] = sm
        return res
        
        res = [None, 0]
        for r in range(0, len(mat)):
            sm = 0
            for c in range(0, len(mat[r])):
                sm += mat[r][c]
            if res[0] == None or sm > res[1]:
                res[0] = r
                res[1] = sm
        return res